src/Windows/Avalonia.Win32/Composition/CompositionHost.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System;
+using System.Runtime.InteropServices;
+using Avalonia.OpenGL.Angle;
+using Windows.UI.Composition;
+using Windows.UI.Composition.Interop;
+using WinRT;
+
+namespace Avalonia.Win32
+{
+    class CompositionHost
+    {
+        internal enum DISPATCHERQUEUE_THREAD_APARTMENTTYPE
+        {
+            DQTAT_COM_NONE = 0,
+            DQTAT_COM_ASTA = 1,
+            DQTAT_COM_STA = 2
+        };
+
+        internal enum DISPATCHERQUEUE_THREAD_TYPE
+        {
+            DQTYPE_THREAD_DEDICATED = 1,
+            DQTYPE_THREAD_CURRENT = 2,
+        };
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct DispatcherQueueOptions
+        {
+            public int dwSize;
+
+            [MarshalAs(UnmanagedType.I4)]
+            public DISPATCHERQUEUE_THREAD_TYPE threadType;
+
+            [MarshalAs(UnmanagedType.I4)]
+            public DISPATCHERQUEUE_THREAD_APARTMENTTYPE apartmentType;
+        };
+
+        [DllImport("coremessaging.dll", EntryPoint = "CreateDispatcherQueueController", CharSet = CharSet.Unicode)]
+        internal static extern IntPtr CreateDispatcherQueueController(DispatcherQueueOptions options, out IntPtr dispatcherQueueController);
+
+        public static CompositionHost Instance { get; } = new CompositionHost();
+
+        private Compositor _compositor;
+        private Windows.System.DispatcherQueueController _dispatcherQueueController;
+        private CompositionGraphicsDevice _graphicsDevice;
+
+        private CompositionHost()
+        {
+            Initialize();
+        }
+
+        private void Initialize()
+        {
+            EnsureDispatcherQueue();
+            if (_dispatcherQueueController != null)
+                _compositor = new Compositor();
+
+            var interop = _compositor.As<ICompositorInterop>();
+
+            var display = Win32GlManager.EglFeature.Display as AngleWin32EglDisplay;
+
+            _graphicsDevice = interop.CreateGraphicsDevice(display.GetDirect3DDevice());
+        }
+
+        public ICompositionDrawingSurfaceInterop InitialiseWindowCompositionTree(IntPtr hwnd, out Windows.UI.Composition.Visual surfaceVisual)
+        {
+            var target = CreateDesktopWindowTarget(hwnd);
+
+            var surface = _graphicsDevice.CreateDrawingSurface(new Windows.Foundation.Size(0, 0),
+                Windows.Graphics.DirectX.DirectXPixelFormat.B8G8R8A8UIntNormalized,
+                Windows.Graphics.DirectX.DirectXAlphaMode.Premultiplied);
+
+            var surfaceInterop = surface.As<ICompositionDrawingSurfaceInterop>();
+
+            var brush = _compositor.CreateSurfaceBrush(surface);
+            var visual = _compositor.CreateSpriteVisual();
```

Comment:
CreateSpriteVisual is available since Windows 10.0.10586.
And CreateBackdropBrush below - since 10.0.14393.

Additional checks is required to avoid exceptions. Or do not allow WinUI.Composition backend for older Win10 versions.